LogMonitor/docs/README.md

```diff
@@ -251,3 +252,101 @@ CMD "c:\\windows\\system32\\ping.exe -n 20 localhost"
+
+### Custom Log Format Pattern Layout
+To ensure the different field values are correctly displayed in the customized log outputs, ensure to wrap the field names within modulo operators (%) and the field names specified matches the correct log sources' field names.
+
+For example: `%Message%, %TimeStamp%`<br>
+
+Each log source tracked by log monitor <em>(ETW, Log File and Event logs)</em> has log field names specific to them:
+
+<strong>Event Logs:</strong>
+  - `Source`: The log source (Event Log)
+  - `TimeStamp`: Time at which the event was generated
+  - `EventID`: Unique identifier assigned to an individual event
+  - `Severity`: A label that indicates the importance or criticality of an event
+  - `Message`: The event message
+
+<strong>ETW:</strong>
+  - `Source`: The log source (ETW)
+  - `TimeStamp`: Time at which the event was generated
+  - `Severity`: A label that indicates the importance or criticality of an event
+  - `ProviderId`:  Unique identifier that is assigned to the event provider during its registration process.
+  - `ProviderName`: Unique identifier or name assigned to an event provider
+  - `DecodingSource`: Component or provider responsible for decoding and translating raw event data into a human-readable format
+  - `ExecutionProcessId`: Identifier associated with a process that is being executed at the time an event is generated
+  - `ExecutionThreadId`: Identifier associated with a thread at the time an event is generated
+  - `Keyword`:  Flag or attribute assigned to an event or a group of related events
+  - `EventId`: Unique identifier assigned to an individual event
+  - `EventData`: Payload or data associated with an event.
+
+<strong>Log Files:</strong>
+  - `Source`: The log source (File)
+  - `TimeStamp`: Time at which the change was introduced in the monitored file.
+  - `FileName`: Name of the file that the log entry is read from.
+  - `Message`: The line/change added in the monitored file.
+
+### Sample Custom Log Configuration
+
+```json
+{
+  "LogConfig": {
+    "logFormat": "custom",
+    "sources": [
+      {
+        "type": "ETW",
+        "eventFormatMultiLine": false,
+        "providers": [
+          {
+            "providerName": "Microsoft-Windows-WLAN-Drive",
+            "providerGuid": "DAA6A96B-F3E7-4D4D-A0D6-31A350E6A445",
+            "level": "Information"
+          }
+        ],
+        "customLogFormat": "{'TimeStamp':'%TimeStamp%', 'source':'%Source%', 'Severity':'%Severity%', 'ProviderId':'%ProviderId%', 'ProviderName':'%ProviderName%', 'EventId':'%EventId%', 'EventData':'%EventData%'}"
+      },
+      {
+        "type": "File",
+        "directory": "c:\\inetpub\\logs",
+        "filter": "*.log",
+        "includeSubdirectories": true,
+        "customLogFormat": "{'message':%Message%,'source':%Source%,'fileName':%FileName%}"
```

Comment:
Is the `customLogFormat` field mandatory or is there a default?

Reply:
There is a default (in case a user doesn't specify)